engine/itest/src/test/java/org/opennms/oce/engine/itest/MockModelBuilder.java

```diff
@@ -0,0 +1,111 @@
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.oce.engine.itest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.opennms.oce.model.api.Model;
+import org.opennms.oce.model.impl.ModelBuilderImpl;
+import org.opennms.oce.model.v1.schema.Inventory;
+import org.opennms.oce.model.v1.schema.MetaModel;
+import org.opennms.oce.model.v1.schema.ModelObjectEntry;
+import org.opennms.oce.model.v1.schema.PeerRef;
+import org.opennms.oce.model.v1.schema.RelativeRef;
+
+public class MockModelBuilder {
+    private final List<ModelObjectEntry> modelObjects = new ArrayList<>();
+
+    public MockModelBuilder withModelObject(String type, String id) {
+        return withModelObject(type, id, ModelBuilderImpl.MODEL_ROOT_TYPE, ModelBuilderImpl.MODEL_ROOT_ID);
+    }
+
+    public MockModelBuilder withModelObject(String type, String id, String parentType, String parentId) {
+        final ModelObjectEntry moe = new ModelObjectEntry();
+        moe.setType(type);
+        moe.setId(id);
+        moe.setParentType(parentType);
+        moe.setParentId(parentId);
+        modelObjects.add(moe);
+        return this;
+    }
+
+    public MockModelBuilder withPeerRelation(String type, String id, String typeA, String idA, String typeZ, String idZ) {
+        final ModelObjectEntry moe = getMoById(type, id);
+
+        PeerRef peerRefA = new PeerRef();
+        peerRefA.setType(typeA);
+        peerRefA.setId(idA);
+        peerRefA.setEndpoint("A");
+        moe.getPeerRef().add(peerRefA);
+
+        PeerRef peerRefZ = new PeerRef();
+        peerRefZ.setType(typeZ);
+        peerRefZ.setId(idZ);
+        peerRefZ.setEndpoint("Z");
+        moe.getPeerRef().add(peerRefZ);
+
+        return this;
+    }
+
+    public MockModelBuilder withRelativeRelation(String type, String id, String relativeType, String relativeId) {
+        final ModelObjectEntry moe = getMoById(type, id);
+
+        RelativeRef relativeRef = new RelativeRef();
+        relativeRef.setType(relativeType);
+        relativeRef.setId(relativeId);
+        moe.getRelativeRef().add(relativeRef);
+
+        return this;
+    }
+
+    private ModelObjectEntry getMoById(String type, String id) {
+        return modelObjects.stream()
+                .filter(mo -> type.equals(mo.getType()) && id.equals(mo.getId()))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(String.format("Could not find element with type: %s and id: %s", type, id)));
+    }
+
+    public Model build() {
+        final MetaModel metaModel = new MetaModel();
+        final Inventory inventory = new Inventory();
+
+        final ModelObjectEntry rootEntry = new ModelObjectEntry();
+        rootEntry.setType(ModelBuilderImpl.MODEL_ROOT_TYPE);
+        rootEntry.setId(ModelBuilderImpl.MODEL_ROOT_ID);
+        inventory.getModelObjectEntry().add(rootEntry);
```

Comment:
Thinking about updating inventory: do we care about Inventory object after we build the initial model ( modelBuilder#buildModel) or we can delegate any further changes to Model?

Reply:
We probably shouldn't care about the original inventory object since it's all captured in the actual model we built.

It might be best to delegate changes to the Model itself by providing a series of "updates".